met-web/src/components/engagement/form/EngagementWidgets/Documents/FileDrawer.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useContext, useState } from 'react';
+import Box from '@mui/material/Box';
+import Drawer from '@mui/material/Drawer';
+import Divider from '@mui/material/Divider';
+import { Grid, MenuItem } from '@mui/material';
+import { MetHeader3, MetLabel, PrimaryButton, SecondaryButton } from 'components/common';
+import { useForm, FormProvider, SubmitHandler } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import ControlledTextField from 'components/common/ControlledInputComponents/ControlledTextField';
+import { useAppDispatch } from 'hooks';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { DocumentsContext } from './DocumentsContext';
+import ControlledSelect from 'components/common/ControlledInputComponents/ControlledSelect';
+import { postDocument } from 'services/widgetService/DocumentService.tsx';
+
+const schema = yup
+    .object({
+        name: yup.string().required(),
+        link: yup.string().required(),
+        folderId: yup.number(),
+    })
+    .required();
+
+type ContactForm = yup.TypeOf<typeof schema>;
+
+const FileDrawer = () => {
+    const dispatch = useAppDispatch();
+    const { documents, handleFileDrawerOpen, fileDrawerOpen, widget } = useContext(DocumentsContext);
+
+    const [isCreatingFile, setIsCreatingDocument] = useState(false);
+
+    const methods = useForm<ContactForm>({
+        resolver: yupResolver(schema),
+    });
+
+    const { handleSubmit } = methods;
+
+    const onSubmit: SubmitHandler<ContactForm> = async (data: ContactForm) => {
+        if (!widget) {
+            return;
+        }
+
+        try {
+            setIsCreatingDocument(true);
+            await postDocument(widget.id, {
+                name: data.name,
+                parent_document_id: data.folderId,
+                document_url: data.link,
```

Comment:
i think backend needs the type as well ; also its not name in backend , its title; But I can change the backend to name if thats better

Reply:
`{
    "title":"1.pdf",
    "type":"file",
    "parent_document_id": 7
}`
this is one sample request

Reply:
done, thank you